blockstack-vm/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+pub mod types;
+pub mod representations;
+pub mod parser;
+pub mod contexts;
+pub mod errors;
+pub mod database;
+
+mod functions;
+
+use types::{Value, CallableType};
+use representations::SymbolicExpression;
+use contexts::{Context, Environment};
+use functions::define::DefineResult;
+use errors::Error;
+
+type InterpreterResult = Result<Value, Error>;
+
+fn lookup_variable(name: &str, context: &Context, env: &Environment) -> InterpreterResult {
+    // first off, are we talking about a constant?
+    if name.starts_with(char::is_numeric) {
+        match i128::from_str_radix(name, 10) {
```

Comment:
Do we want to support alternative radixes?  Like, say, base-16?

Reply:
Maybe -- I was planning on lexing hexstrings into buffers, rather than ints. We could support both. Will create an issue for it.